File: Java/EX14.java
import java.util.Properties;

class Configuracao {
    // Atributo estático para armazenar a instância única da classe
    private static Configuracao instancia;

    // Atributo para armazenar as configurações
    private final Properties configuracao;

    // Construtor privado para evitar a criação de instâncias externas
    private Configuracao() {
        configuracao = new Properties(); // Inicializa as configurações
    }

    // Método estático para obter a instância única da classe
    public static Configuracao getInstancia() {
        if (instancia == null) { // Verifica se a instância já foi criada
            instancia = new Configuracao();
        }
        return instancia;
    }

    // Método para definir uma configuração
    public void setConfig(String chave, String valor) {
        configuracao.setProperty(chave, valor);
    }

    // Método para obter uma configuração
    public String getConfig(String chave) {
        return configuracao.getProperty(chave);
    }
}

// Exemplo de uso da classe Configuracao
class Main7 {
    public static void main(String[] args) {
        // Obtendo a instância única da classe Configuracao
        Configuracao config1 = Configuracao.getInstancia();
        Configuracao config2 = Configuracao.getInstancia();

        // Definindo configurações
        config1.setConfig("url", "https://example.com");
        config1.setConfig("timeout", "30");

        // Obtendo configurações usando a segunda instância
        System.out.println("URL: " + config2.getConfig("url")); // Saída: URL: https://example.com
        System.out.println("Timeout: " + config2.getConfig("timeout")); // Saída: Timeout: 30

        // Verificando se ambas as instâncias são a mesma
        System.out.println("config1 é config2? " + (config1 == config2)); // Saída: config1 é config2? true
    }
}

File: Python/EX14.py
class Configuracao:
    _instancia = None  # Atributo de classe para armazenar a instância única

    def __new__(cls, *args, **kwargs):
        if cls._instancia is None:  # Verifica se a instância já foi criada
            cls._instancia = super(Configuracao, cls).__new__(cls)
            # Aqui você pode inicializar qualquer atributo da instância
            cls._instancia.configuracao = {}
        return cls._instancia

    def set_config(self, chave, valor):
        self.configuracao[chave] = valor

    def get_config(self, chave):
        return self.configuracao.get(chave, None)

# Exemplo de uso
if __name__ == "__main__":
    # Criando instâncias da classe Configuracao
    config1 = Configuracao()
    config2 = Configuracao()

    # Definindo configurações
    config1.set_config("url", "https://example.com")
    config1.set_config("timeout", 30)

    # Obtendo configurações usando a segunda instância
    print(f"URL: {config2.get_config('url')}")  # Saída: URL: https://example.com
    print(f"Timeout: {config2.get_config('timeout')}")  # Saída: Timeout: 30

    # Verificando se ambas as instâncias são a mesma
    print(f"config1 é config2? {config1 is config2}")  # Saída: config1 é config2? True
